### PR TITLE
Move logical-line rules to `token` rules

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

Move the logical line rules into the token linting phase.

## Test Plan

`cargo test`

### CODE REVIEW

**Comment by MichaReiser** [comment]
Current dependencies on/for this PR:
* main
  * **PR #7654** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/7654" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/7654?utm_source=stack-comment).

File: crates/ruff_linter/src/linter.rs [hunk 1]
```diff
@@ -119,17 +120,6 @@ pub fn check_path(
         diagnostics.extend(check_file_path(path, package, settings));
     }
 
-    // Run the logical line-based rules.
-    if settings
-        .rules
-        .iter_enabled()
-        .any(|rule_code| rule_code.lint_source().is_logical_lines())
```
Comment (charliermarsh) [inline crates/ruff_linter/src/linter.rs:126]: I think you now need to add `is_logical_lines` to line 97.

**Review by charliermarsh — COMMENTED** [review]
Not necessarily disagreeing, but what's the motivation?

**Comment by MichaReiser** [comment]
It makes it clearer how many lint phases we have and we could explore changing all token based lints to be invoked per token rather than each driving their own iteration

**Review by charliermarsh — APPROVED** [review]

**Pull request closed without merging**